packages/react-server/core/ClientController.js

```diff
@@ -88,7 +96,23 @@ class ClientController extends EventEmitter {
 			var url = request.getUrl();
 
 			if (type === History.events.PUSHSTATE) {
-				this._history.pushState({frameback:true}, null, url);
+				// Sorry folks.  If we need to do a client
+				// transition, then we're going to clobber
+				// your state.  You must be able to render
+				// from URL, anyway, so if you're set up right
+				// it won't affect user experience.  It means,
```

Comment:
Yes, it affects performance.  If client code behaves correctly, though, it won't affect content or behavior.